qp/utils.py

```diff
@@ -0,0 +1,46 @@
+import numpy as np
+import sys
+
+def safelog(arr, threshold=sys.float_info.epsilon):
+    """
+    Takes log of array with zeroes.
+
+    Parameters
+    ----------
+    arr: ndarray
+        Values to be logged
+    threshold: float
+        Small, positive value to replace zeros and negative numbers
+
+    Returns
+    -------
+    logged: ndarray
+        Logged values, with approximation in place of zeros and negative numbers
+    """
+    shape = np.shape(arr)
+    flat = arr.flatten()
+    logged = np.log(np.array([max(a,threshold) for a in flat])).reshape(shape)
+    return logged
+
+def calckl(qn, pn):
+    """
+    Calculates Kullback-Leibler Divergence
+
+    Parameters
+    ----------
+    qn: ndarray
```

Comment:
I think the case you propose may not be appropriate for the KLD.  Because of the KLD's asymmetry, there has to be a notion of Q being conditioned on P.  When neither is more of a truth than the other, I don't know how to interpret it.  I did implement the RMS, which may be better suited to this use case.  Thoughts?